### PR TITLE
Spread page

### LINKED ISSUE #52: Pages > spread page
spread

### PR DESCRIPTION
**Pull request by JanaKhanji**

### CODE REVIEW

File: src/Pages/SpreadPage/SpreadPage.js [hunk 1]
```diff
@@ -1,4 +1,83 @@
...
+  function showevent(date) {
+    setmonthEvents({
+      month: months[date.getMonth()],
+      events: [
+        ...events.filter(
+          (el) => new Date(el.date).getMonth() === date.getMonth()
+        ),
+      ],
+    });
+    console.log(monthEvents);
```
Comment (osamaakb) [inline src/Pages/SpreadPage/SpreadPage.js]: can you remove this log

File: src/Pages/SpreadPage/SpreadPage.js [hunk 2]
```diff
@@ -1,4 +1,83 @@
...
+
+  function CalendarFct(date) {
+    let todayEvents = events.find((el) => el.date === date.toDateString());
+    if (date.toDateString() === new Date().toDateString()) return 'today';
+    else if (todayEvents) {
+      return EventsType.find((el) => el.id === todayEvents.type_id).name;
+    }
+  }
+
+  function showevent(date) {
```
Comment (osamaakb) [inline src/Pages/SpreadPage/SpreadPage.js:41]: is this for the current year only? what if we want to add events for next year or had events in the previous years? 
or do you want it to be only for the current year?

File: src/Pages/SpreadPage/SpreadPage.js [hunk 3]
```diff
@@ -1,4 +1,83 @@
...
+          startDate={new Date(year, i, 1)}
+          classDayfct={CalendarFct}
+          onClick={showevent}
+        />
+      </div>
+    );
+  }
+
+  return (
+    <>
```
Comment (osamaakb) [inline src/Pages/SpreadPage/SpreadPage.js:70]: you don't need the fragment here, you already have a div

File: src/Components/SpreadModals/EventsModal.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import EventsType from '../../Constants/EventsType.js';
+
+export default function EventsModal({ setShowModal, monthEvents }) {
+  return (
+    <>
```
Comment (osamaakb) [inline src/Components/SpreadModals/EventsModal.js:5]: No need for the fragment here

File: src/Components/SpreadModals/EventForm.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      id: new Date().getTime(),
+      title: title.current.value,
+      description: description.current.value,
+      date: new Date(date.current.value).toDateString(),
+      type: type,
+    };
+    console.log(obj);
+  }
+  return (
+    <>
```
Comment (osamaakb) [inline src/Components/SpreadModals/EventForm.js:25]: No need for the fragment here

**Review by osamaakb — CHANGES_REQUESTED** [review]

File: src/Pages/SpreadPage/SpreadPage.js (hunk 2 above)
Comment (JanaKhanji) [reply src/Pages/SpreadPage/SpreadPage.js:41]: let us keep it only for the current year

File: src/Components/SpreadModals/EventsModal.js (hunk 4 above)
Comment (JanaKhanji) [reply src/Components/SpreadModals/EventsModal.js:5]: i need it here cz i have two divs one for the modal and one for the gray background

File: src/Pages/SpreadPage/SpreadPage.js (hunk 3 above)
Comment (JanaKhanji) [reply src/Pages/SpreadPage/SpreadPage.js:70]: this wont work without <> because the modals have height screen

**Review by osamaakb — APPROVED** [review]

**Pull request merged**